subsys/bluetooth/services/ias/ias_client.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <errno.h>
+#include <init.h>
+#include <stdint.h>
+#include <zephyr.h>
+#include <net/buf.h>
+#include <sys/check.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/services/ias.h>
+
+#include "ias_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_IAS_CLIENT)
+#define LOG_MODULE_NAME bt_ias_client
+#include "../../common/log.h"
+
+#define IS_HANDLE_VALID(handle) ((handle) != 0x0000)
+
+static struct bt_ias_client_cb *ias_client_cb;
+static struct bt_ias_client client_list[CONFIG_BT_MAX_CONN];
+
+static struct bt_ias_client *client_by_conn(struct bt_conn *conn)
+{
+	return &client_list[bt_conn_index(conn)];
+}
+
+int bt_ias_alert_write(struct bt_conn *conn, enum bt_ias_alert_lvl alert_lvl)
+{
+	int err;
+	uint8_t lvl = (uint8_t)alert_lvl;
```

Comment:
```suggestion
	const uint8_t lvl = (uint8_t)alert_lvl;
```

Reply:
The alternative here is to use `__packed` on the `enum bt_ias_alert_lvl` so that it is forced to be an octet itself

Reply:
I've changed this enum so it makes use of __packed now.